src/decode.rs

```diff
@@ -17,22 +17,31 @@
 // WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 // CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-quick_error! {
-    #[derive(Debug, PartialEq, Eq)]
-    pub enum Error {
-        /// Not enough input bytes.
-        Insufficient {
-            display("not enough input bytes")
-        }
-        /// Input bytes exceed maximum.
-        Overflow {
-            display("input bytes exceed maximum")
+use std::{self, fmt};
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Error {
+    /// Not enough input bytes.
+    Insufficient,
+    /// Input bytes exceed maximum.
+    Overflow,
+
+    #[doc(hidden)]
+    __Nonexhaustive
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::Insufficient => f.write_str("not enough input bytes"),
+            Error::Overflow => f.write_str("input bytes exceed maximum"),
+            _ => f.write_str("__Nonexhaustive")
```

Comment:
I'd suggest putting `__Nonexhaustive` there in order to not forget to add new variants.